dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java

```diff
@@ -92,114 +90,80 @@ public class EE10FeatureCompatibilityTest extends FATServletClient {
     @Server("jakartaee10.fat")
     public static LibertyServer server;
 
-    static {
-        nonEE10JavaEEFeatures.addAll(EE7FeatureReplacementAction.EE7_FEATURE_SET);
-        nonEE10JavaEEFeatures.addAll(EE8FeatureReplacementAction.EE8_FEATURE_SET);
-
-        // appSecurity-1.0 is superceded by appSecurity-2.0 so it isn't in one of the replacement
-        // feature lists.  jaxb and jaxws 2.3 are EE related, but are noship features currently.
-        // jsp-2.2 is a EE6 feature that is included with open liberty.
-        // websocket-1.0 is a special case.  Part of EE7, but 1.1 is used by liberty.
-        nonEE10JavaEEFeatures.add("appSecurity-1.0");
-        nonEE10JavaEEFeatures.add("jsp-2.2");
-        nonEE10JavaEEFeatures.add("websocket-1.0");
-
-        // Remove test features that are in the FeatureReplacementActions
-        nonEE10JavaEEFeatures.remove("componenttest-1.0");
-        nonEE10JavaEEFeatures.remove("componenttest-2.0");
-        nonEE10JavaEEFeatures.remove("txtest-1.0");
-        nonEE10JavaEEFeatures.remove("txtest-2.0");
-        nonEE10JavaEEFeatures.remove("ejbTest-1.0");
-        nonEE10JavaEEFeatures.remove("ejbTest-2.0");
-
-        for (FeatureSet mpFeatureSet : MicroProfileActions.ALL) {
-            if (mpFeatureSet.getEEVersion() != EEVersion.EE10) {
-                nonEE10MicroProfileFeatures.addAll(mpFeatureSet.getFeatures());
-            }
+    static Set<String> getAllCompatibleFeatures() {
+        Set<String> allFeatures = new HashSet<>();
+        try {
+            File installRoot = new File(Bootstrap.getInstance().getValue("libertyInstallPath"));
+            allFeatures.addAll(FeatureUtilities.getFeaturesFromServer(installRoot));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
         }
+        return getCompatibleFeatures(allFeatures);
+    }
 
-        // MP standalone features
-        for (FeatureSet mpFeatureSet : MicroProfileActions.STANDALONE_ALL) {
-            if (mpFeatureSet.getEEVersion() != EEVersion.EE10) {
-                nonEE10MicroProfileFeatures.addAll(mpFeatureSet.getFeatures());
-            }
-        }
+    private static Set<String> getCompatibleFeatures(Set<String> allFeatures) {
+        Set<String> compatibleFeatures = new HashSet<>();
+
+        // By default, features are assumed to be compatible
+        compatibleFeatures.addAll(allFeatures);
+
+        // Non-ee10 features are not compatible
+        compatibleFeatures.removeAll(FeatureUtilities.allEeFeatures());
+        compatibleFeatures.addAll(JakartaEE10Action.EE10_FEATURE_SET);
+
+        // MP features are only compatible if they're in MP versions which work with EE10
+        compatibleFeatures.removeAll(FeatureUtilities.allMpFeatures());
+        // Add only the MP 6.0 features which we've made compatible
+        compatibleFeatures.add("mpConfig-3.0");
```

Comment:
I believe with you adding mpConfig-3.0 being compatible mpRestClient-3.0 should also work.

Reply:
That's possibly true, I hadn't added it here because that wasn't something I was trying to make work, but as you've pointed it out I can add it now.

Reply:
oh ok.  That is easy enough.